tests/test_projectbuilder.py

```diff
@@ -148,26 +146,53 @@ def test_check_dependencies(mocker):
         not builder.check_dependencies('wheel')
 
 
-@pytest.mark.skipif(sys.version_info[:2] == (3, 5), reason='bug in mock')
-def test_build(mocker):
-    open_mock = mocker.mock_open(read_data=DUMMY_PYPROJECT)
+def test_build(mocker, pyproject_mock):
     mocker.patch('importlib.import_module')
-    mocker.patch('{}.open'.format(build_open_owener), open_mock)
+    mocker.patch('{}.open'.format(build_open_owner), pyproject_mock)
     mocker.patch('pep517.wrappers.Pep517HookCaller')
 
-    builder = build.ProjectBuilder('.')
+    builder = build.ProjectBuilder()
 
     builder.hook.build_sdist.side_effect = [None, Exception]
     builder.hook.build_wheel.side_effect = [None, Exception]
 
     builder.build('sdist', '.')
-    builder.hook.build_sdist.assert_called()
+    builder.hook.build_sdist.assert_called_with('.', {})
 
     builder.build('wheel', '.')
-    builder.hook.build_wheel.assert_called()
+    builder.hook.build_wheel.assert_called_with('.', {})
 
     with pytest.raises(build.BuildBackendException):
         builder.build('sdist', '.')
 
     with pytest.raises(build.BuildBackendException):
         builder.build('wheel', '.')
+
+
+def test_default_backend(mocker, empty_file_mock):
+    mocker.patch('importlib.import_module')
+    mocker.patch('{}.open'.format(build_open_owner), empty_file_mock)
+    mocker.patch('pep517.wrappers.Pep517HookCaller')
+
+    builder = build.ProjectBuilder()
+
+    assert builder._build_system == build._DEFAULT_BACKEND
+
+
+def test_missing_backend(mocker, pyproject_no_backend_mock):
+    mocker.patch('importlib.import_module')
+    mocker.patch('{}.open'.format(build_open_owner), pyproject_no_backend_mock)
+    mocker.patch('pep517.wrappers.Pep517HookCaller')
```

Comment:
Why is this mocked out? Because the constructor will try and import the build backend? If so, can we try a more targeted mock that would simply *provide* the build backend?

Reply:
Yes, this can be removed when we refactor the test suite to use proper packages.